src/sessions.ts

```diff
@@ -594,7 +592,7 @@ function userExplicitlyEndedTransaction(session: ClientSession) {
   return USER_EXPLICIT_TXN_END_STATES.has(session.transaction.state);
 }
 
-function attemptTransaction<T>(
+async function attemptTransaction<T>(
   session: ClientSession,
   startTime: number,
   fn: WithTransactionCallback<T>,
```

Comment:
I can't leave a comment on it but can we change this to return `Promise<T>` instead, it is just a correctness improvement.